packages/esm-ward-app/src/ward-workspace/admit-patient-form-workspace/admit-patient-form-workspace.workspace.tsx

```diff
@@ -0,0 +1,174 @@
+  ExtensionSlot,
+  usePatient,
+  type Patient,
+  useConfig,
+  showSnackbar,
+} from '@openmrs/esm-framework';
+import styles from './admit-patient-form.scss';
+import { useAdmissionLocation } from '../../hooks/useAdmissionLocation';
+import useWardLocation from '../../hooks/useWardLocation';
+import { useTranslation } from 'react-i18next';
+import { filterBeds } from '../../ward-view/ward-view.resource';
+import type { BedLayout } from '../../types';
+import { z } from 'zod';
+import { Controller, useForm } from 'react-hook-form';
+import { zodResolver } from '@hookform/resolvers/zod';
+import { Form, Row, Column, DropdownSkeleton, Dropdown, ButtonSet, Button } from '@carbon/react';
+import { InlineNotification } from '@carbon/react';
+import { assignPatientToBed, createEncounter } from '../../ward.resource';
+import type { WardConfigObject } from '../../config-schema';
+import { useInpatientRequest } from '../../hooks/useInpatientRequest';
+
+interface AdmitPatientFormWorkspaceProps extends DefaultWorkspaceProps {
+  patient: Patient;
+}
+
+const AdmitPatientFormWorkspace: React.FC<AdmitPatientFormWorkspaceProps> = ({
+  patient,
+  closeWorkspace,
+  closeWorkspaceWithSavedChanges,
+  promptBeforeClosing,
+}) => {
+  const { t } = useTranslation();
+  const { location } = useWardLocation();
+  const [isSubmitting, setIsSubmitting] = useState(false);
+  const { mutate: mutateInpatientRequest } = useInpatientRequest();
+  const { admissionEncounterTypeUuid } = useConfig<WardConfigObject>();
+  const [showErrorNotifications, setShowErrorNotifications] = useState(false);
+  const { isLoading, admissionLocation, mutate: mutateAdmissionLocation } = useAdmissionLocation();
+  const beds = useMemo(() => (isLoading ? [] : filterBeds(admissionLocation)), [admissionLocation]);
+  const getBedRepresentation = useCallback((bedLayout: BedLayout) => {
+    const bedNumber = bedLayout.bedNumber;
+    const patients =
+      bedLayout.patients.length === 0
+        ? [t('emptyBed', 'Empty')]
+        : bedLayout.patients.map((patient) => patient?.person?.preferredName?.display);
+    return [bedNumber, ...patients].join(' · ');
+  }, []);
+
+  const zodSchema = useMemo(
+    () =>
+      z.object({
+        bedId: z.number({
+          required_error: t('bedSelectionRequired', 'Please select a bed to move forward'),
+        }),
+      }),
+    [],
+  );
+
+  type FormValues = z.infer<typeof zodSchema>;
+
+  const {
+    control,
+    formState: { errors, isDirty },
+    handleSubmit,
+    getValues,
+    watch,
+  } = useForm<FormValues>({
+    resolver: zodResolver(zodSchema),
+  });
+
+  useEffect(() => {
+    promptBeforeClosing(() => isDirty);
+  }, [isDirty]);
+
+  const onSubmit = useCallback(
+    (values: FormValues) => {
+      setShowErrorNotifications(false);
+      setIsSubmitting(true);
+      const bedSelected = beds.find((bed) => bed.bedId === values.bedId);
+      createEncounter(patient.uuid, admissionEncounterTypeUuid, location?.uuid)
```

Comment:
We should fetch the encounter type from the configuration endpoint (see comment) above, and also we should use the "admission" encounter type if the inpatient request was a of type "ADMIT", and the "transferWithinHospital" encounter type if the inpatient request was of type "TRANSFER"... (I don't think we have access to this information currently in this view so we may need to pass it in as a prop)

Reply:
The transfer form is a different form. This form is just made for Admission. The transfer form is covered in O3-3247, which is separate from this workflow.
Thanks!